### PR TITLE
feat: [PL-32366]: Multiple IDP support

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
### Summary

After this change User will be able to configure multiple SAML Providers to their account. 

Note - I have used existing saml provider code and broken down into multiple steps. Created separate V2 components, so that the old and new code shouldn't be mixed. 
All the CSS properties and logic is taken from the existing component.

#### Screenshots

---- Before -------
Screenshot 2023-05-08 at 7 22 41 PM

Screenshot 2023-05-08 at 7 22 30 PM

---- After ------

Screenshot 2023-05-08 at 7 23 58 PM

Screenshot 2023-05-08 at 7 23 46 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by pbarapatre10 — APPROVED** [review]

**Review by sunnykesh-harness — APPROVED** [review]

File: src/modules/33-auth-settings/modals/SAMLProvider/useSAMLProviderModalV2.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+interface UseSAMLProviderReturn {
+  openSAMlProvider: (_samlProvider?: SAMLSettings) => void
+  closeSAMLProvider: () => void
+}
+
+export const useSAMLProviderModalV2 = ({ onSuccess }: Props): UseSAMLProviderReturn => {
+  const { getString } = useStrings()
+  const [samlProvider, setSamlProvider] = React.useState<SAMLSettings>()
+  const [showModal, hideModal] = useModalHook(
```
Comment (abhinavrastogi-harness) [inline src/modules/33-auth-settings/modals/SAMLProvider/useSAMLProviderModalV2.tsx:32]: We should use ModalDialog component instead of useModal hook.

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/33-auth-settings/modals/SAMLProvider/steps/IdentityProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                <FormInput.Text name="logoutUrl" label={getString('authSettings.logoutUrl')} />
+              </Container>
+            </Layout.Vertical>
+            <Layout.Horizontal padding={{ top: 'small' }} spacing="medium">
+              <Button
+                text={getString('back')}
+                icon="chevron-left"
+                variation={ButtonVariation.SECONDARY}
+                onClick={() => props?.previousStep?.({ ...props.prevStepData } as IdentityProviderFormValues)}
+                data-name="awsBackButton"
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/modals/SAMLProvider/steps/IdentityProvider.tsx:99]: Could simply be 'backButton'

File: src/modules/33-auth-settings/pages/Configuration/SAMLProvider/SAMLProviderV2.tsx [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+  })
+
+  React.useEffect(() => {
+    setUpdating(updatingAuthMechanismToSaml || deletingSamlSettings || loadingEnableDiableAuthSettings)
+  }, [updatingAuthMechanismToSaml, deletingSamlSettings, loadingEnableDiableAuthSettings, setUpdating])
+
+  const { openDialog: confirmSamlSettingsDelete } = useConfirmationDialog({
+    titleText: getString('authSettings.deleteSamlProvider'),
+    contentText: (
+      <String stringID="authSettings.deleteSamlProviderDescription" useRichText={true} vars={{ displayName: '' }} />
```
Comment (mayankVermaHarness) [inline src/modules/33-auth-settings/pages/Configuration/SAMLProvider/SAMLProviderV2.tsx:139]: `displayName` is always an empty string here. Is this expected?
The string is:
`'Do you want to delete the SAML Provider: <strong> {{displayName}} </strong> ?'`

**Review by mayankVermaHarness — APPROVED** [review]

File: src/modules/33-auth-settings/pages/Configuration/SAMLProvider/SAMLProviderV2.tsx (hunk 3 above)
Comment (avinashmadhwani02) [reply src/modules/33-auth-settings/pages/Configuration/SAMLProvider/SAMLProviderV2.tsx]: Done

File: src/modules/33-auth-settings/modals/SAMLProvider/useSAMLProviderModalV2.tsx (hunk 1 above)
Comment (avinashmadhwani02) [reply src/modules/33-auth-settings/modals/SAMLProvider/useSAMLProviderModalV2.tsx:32]: I have used the existing code here, haven't modified any logic.

File: src/modules/33-auth-settings/modals/SAMLProvider/steps/IdentityProvider.tsx (hunk 2 above)
Comment (avinashmadhwani02) [reply src/modules/33-auth-settings/modals/SAMLProvider/steps/IdentityProvider.tsx:99]: Using this code from existing component.

**Review by sunnykesh-harness — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**